backend/hyper/entity_test.go

```diff
@@ -124,3 +134,175 @@ func TestEntityMutation_Drafts(t *testing.T) {
+	require.NoError(t, carolBlobs.SaveBlob(ctx, chD))
+	require.NoError(t, carolBlobs.SaveBlob(ctx, chE))
+	carolView, err := carolBlobs.LoadEntity(ctx, "foo")
+	require.NoError(t, err)
+	require.Len(t, carolView.heads, 2)
+
+	require.Equal(t, map[cid.Cid]struct{}{chE.CID: {}, chC.CID: {}}, carolView.heads, "heads must contain both Carol latest changes and Alice's")
+
+	// Carol now uses changes from both Alice an the old version of the document Carol had.
+	chF, err := carolView.CreateChange(carolView.NextTimestamp(), carol.Device, kdCarolBlob.CID, map[string]any{
+		"Last Carol change": "F is untrusted from Alice's perspective",
+	})
+	require.NoError(t, err)
+	require.Contains(t, chF.Decoded.(Change).Deps, chE.CID, "Changes must depend on both heads")
+	require.Contains(t, chF.Decoded.(Change).Deps, chC.CID, "Changes must depend on both heads")
+	require.NoError(t, carolBlobs.SaveBlob(ctx, chF))
+	carolLatestChanges, err := carolBlobs.LoadEntity(ctx, "foo")
+	require.NoError(t, err)
+
+	// Alice now receives Carol changes
+	require.NoError(t, aliceBlobs.SaveBlob(ctx, chF))
+	_, err = aliceBlobs.LoadEntity(ctx, "foo")
+	require.Error(t, err, "missing dependency for change C")
+	require.NoError(t, aliceBlobs.SaveBlob(ctx, chC))
+	aliceView, err := aliceBlobs.LoadEntity(ctx, "foo")
+	require.NoError(t, err)
+	require.Equal(t, carolLatestChanges.heads, aliceView.heads)
+
+	// Alice Creates another entity to check we can later get only foo changes
+	e2 := NewEntity("bar")
+	chG, err := e2.CreateChange(e2.NextTimestamp(), alice.Device, kdAliceBlob.CID, map[string]any{
+		"Alice's New unrelated entity": "G is trusted from Alice's perspective",
+	})
+	require.NoError(t, err)
+	require.NoError(t, aliceBlobs.SaveBlob(ctx, chG))
+
+	// Now we get the trusted entity foo
+	trustedEntity, err := aliceBlobs.LoadTrustedEntity(ctx, "foo")
+	require.NoError(t, err)
+
+	require.Len(t, trustedEntity.heads, 1)
+	require.Equal(t, map[cid.Cid]struct{}{chE.CID: {}}, trustedEntity.heads, "Last trusted change was E made by Alice herself")
+
+	// Now trust Carol and see if we get F changes
+	require.NoError(t, aliceBlobs.SetAccountTrust(ctx, carol.Account.Principal()))
+	trustedEntity, err = aliceBlobs.LoadTrustedEntity(ctx, "foo")
+	require.NoError(t, err)
+	require.Len(t, trustedEntity.heads, 1)
+	require.Equal(t, map[cid.Cid]struct{}{chF.CID: {}}, trustedEntity.heads, "Last trusted change is now Carol's since Carol is trusted now")
+
+	// Now untrust Carol again and see if we get back E changes
+	require.NoError(t, aliceBlobs.UnsetAccountTrust(ctx, carol.Account.Principal()))
+	trustedEntity, err = aliceBlobs.LoadTrustedEntity(ctx, "foo")
+	require.NoError(t, err)
+	require.Len(t, trustedEntity.heads, 1)
+	require.Equal(t, map[cid.Cid]struct{}{chE.CID: {}}, trustedEntity.heads, "Last trusted change was E made by Alice herself")
+
+	// Ask for an unknown entity
+	fakeEntity, err := aliceBlobs.LoadTrustedEntity(ctx, "fake_entity")
+	require.NoError(t, err)
```

Comment:
Why loading non-existing entity doesn't return a not found error? All the other loading methods do return errors in this situation.

Reply:
@burdiyan Since the entity lookup is in a subquery inside the `ChangesGetTrustedHeadsJSON` SQL, it returns an empty json list `[]` with `nil` error. Then is passed to the `loadFromHeads` function. which takes empty heads and return a nil entity. The difference between the `LoadTrustedEntity` and its general counterpart is that in the latter we do a explicit `edb, err := hypersql.EntitiesLookupID(conn, string(eid))` but in the former that's included in the sql logic.

Reply:
Still, I think it should return an error (could be checked if returns empty array and then fail if so). But I think we should just do the lookup for the internal entity integer ID as a separate query as we do in other loading methods. I did that in my other comment suggesting changes to the SQL query.